File: src/assoc-expo.py
a, b, c = map(int, input().split())

# if((a**(b**c)) != (a**b)**c): print("Oh look, a squirrel!")
# else: print("What an excellent example!")

def power(base, exp, mod):
    r = 1
    while(exp > 0):
        if(exp % 2 == 1):
            r = (r * base) % mod
        base = (base * base) % mod
        exp //= 2
    return r

mod = 1000000000000000000000000000000000000000000000000000000000000000000

if power(a, power(b, c, mod), mod) == power(power(a, b, mod), c, mod): print("What an excellent example!")
else: print("Oh look, a squirrel!")

## a ^ (b ^ c) == (a ^ b)^c

File: src/assoc-expo.cpp
#include<bits/stdc++.h>
using namespace std;

unsigned long long int power(unsigned long long int base, unsigned long long int expo){
    unsigned long long int r = 1;
    while(expo > 0){
        if(expo%2 == 1) r *= base;
        base *= base;
        expo /= 2;
    }
    return r;
}

int main(){
    unsigned long long int a, b, c; cin >> a >> b >> c;
    // if(pow(a, pow(b, c)) != pow(pow(a, b), c)) cout << "Oh look, a squirrel!";
    // else cout << "What an excellent example!";
    if(power(a, power(b, c)) != power(power(a, b), c)) cout << "Oh look, a squirrel!";
    else cout << "What an excellent example!";
    return 0;
}